src/librustc_incremental/calculate_svh.rs

```diff
@@ -74,17 +36,24 @@ impl Svh {
         // to ensure it is not incorporating implementation artifacts into
         // the hash that are not otherwise visible.)
 
+        let crate_disambiguator = self.sess.crate_disambiguator.get();
+        let krate = self.map.krate();
+
         // FIXME: this should use SHA1, not SipHash. SipHash is not built to
         //        avoid collisions.
         let mut state = SipHasher::new();
+        debug!("state: {:?}", state);
 
         "crate_disambiguator".hash(&mut state);
-        crate_disambiguator.len().hash(&mut state);
-        crate_disambiguator.hash(&mut state);
+        crate_disambiguator.as_str().len().hash(&mut state);
```

Comment:
Hmm, ok

On Mon, Apr 04, 2016 at 10:00:47AM -0700, Michael Woerister wrote:
> >  
> >          "crate_disambiguator".hash(&mut state);
> > -        crate_disambiguator.len().hash(&mut state);
> > -        crate_disambiguator.hash(&mut state);
> > +        crate_disambiguator.as_str().len().hash(&mut state);
> 
> >  what I was concerned about is that you cross-compile the library
> 
> That's true. I think, I would just fix it with something like: `(crate_disambiguator.as_str().len() as u64).to_le().hash(...)`
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/rust-lang/rust/pull/32016/files/601f8c66256df0d4d1fe2ec3d3f617a6fae08f68..aa5eca284c2e220cac461a5626a559690f664fe6#r58409485